### DISCUSSION TITLE
Organizations

### DISCUSSION THREAD

**Discussion by esevato** [openobserve/openobserve#985] [Q&A]
Will organizations be added to self hosted or is this going to be reserved for Cloud/Enterprise version only?

This tool is insanely easy to set up, nice work. I've got it on a Proxmox VM and Zimaboard (debian 12) so far, testing LXC next but crazy quick and easy to set up so far.

**Comment by parya19** [comment]
Only thing that is restricting me to implement openobserve is  this organizations  as i need to segregate the data collected from different sources.

**Comment by prabhatsharma** [comment]
@esevato  Organization functionality is available in the APIs in OpenObserve, but missing from the UI. It will be available soon.

@pratibhaaryajan19 You don't really need/want organizations to separate incoming data from different sources. You could use different streams todo that.

e.g. curl http://localhost:5080/api/org_name/stream_name/_json -i -u root@example.com:Complexpass#123  -d "@k8slog_json.json" 

Just change the stream name to whatever you want.

**Comment by oasisk** [comment]
Add User flow also offers opportunity to create organisation , one can use existing user or create new user , choose other in organisation drop down & provide new organisation name. Please find below screenshots :
![image](https://github.com/openobserve/openobserve/assets/35160958/5d4c6667-31bc-472f-9a9e-0c91e81f6987)

**Reply by miversen33** [reply]
~~@oasisk while that is correct, creating a new user and adding them to an Org specified this way just drops the user into the ether. The new org is _not_ created, nor is the user.~~

Disregard, I guess I just needed to clear my browser cache or something. I can see my user and the new org now 🙃

**Reply by wlinna** [reply]
I noticed that the root user cannot see the new organization until it is added there manually

I think the root user should be able to see all organizations by default

**Reply by hengfeiyang** [reply]
root user can see anything after the org have data. `org` is a virtual concept, if there is no data in the org, actually the `org` doesn't exist.

**Reply by wlinna** [reply]
But it is very confusing to not see the organization appear anywhere after creation. I spent a long time reading docs, going through the issue tracker and experimenting if there's a way to make "organizations work". To me it looked like there was an obvious bug that had not been fixed for all this years, and I almost regretted choosing this, until I figured out what I wrote in my previous post.

Also, don't users count as data? I created the other organization with another user as its admin.

**Reply by hengfeiyang** [reply]
there is no org create API, the only way create org is set a new org when you ingest data. for example:

Endpoint: `POST /api/{organization}/{stream}/_json`

Ingest data to `http://localhost:5080/api/org2/default/_json`

You will see there is a `org2`.

https://openobserve.ai/docs/api/ingestion/logs/json/

**Reply by wlinna** [reply]
Are we talking about something different here? I'm trying to tell that the UI where one can create an organization (see the screenshot above) can be very confusing. And the organization is viewable by `root` if `root` has a role in that organization. I had not ingested any data at this point.

**Reply by hengfeiyang** [reply]
i am saying the UI create org is a virtual concpect, just let the user belongs one org, it don't create real org. so root can't see it, because there is no org really created when you create user. the org only created when you ingest data. i am try to explain to you how to let it work today.

We are also working on revamp the org part, but still need  sometime.

**Reply by wlinna** [reply]
Oh, okay. Good to hear that
